EasyPost/Models/Shared/Collection.cs

```diff
@@ -1,15 +1,65 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
 using EasyPost._base;
+using EasyPost.Exceptions.General;
 using Newtonsoft.Json;
 
 namespace EasyPost.Models.Shared
 {
-    public class Collection : EasyPostObject
+    public abstract class Collection : EasyPostObject
     {
         #region JSON Properties
 
         [JsonProperty("has_more")]
         public bool? HasMore { get; set; }
 
         #endregion
+
+        /// <summary>
+        ///     Get the next page of a paginated collection.
+        /// </summary>
+        /// <param name="apiCallFunction">The function to execute to retrieve a page (most likely an All function).</param>
+        /// <param name="currentEntries">The results on the current page. Used to determine the API call parameters to retrieve the next page.</param>
+        /// <typeparam name="T">The type of <see cref="Collection"/> to get the next page of.</typeparam>
+        /// <typeparam name="T2">The type of <see cref="EasyPost._base.EasyPostObject"/> the entries are.</typeparam>
+        /// <returns>The next page of a paginated collection.</returns>
+        /// <exception cref="EndOfPaginationError">Thrown if there is no next page to retrieve.</exception>
+        internal async Task<T> GetNextPage<T, T2>(Func<Dictionary<string, object>, Task<T>> apiCallFunction, List<T2>? currentEntries) where T : Collection where T2 : EasyPost._base.EasyPostObject
```

Comment:
issue: I'm not seeing a way to pass an optional page size of the next page which is part of the acceptance criteria